azure-pipelines.yml

```diff
@@ -11,34 +11,82 @@ pr:
       - develop
   drafts: false
 
-pool:
-  vmImage: ubuntu-latest
-
-steps:
-- task: UseDotNet@2
-  displayName: 'Install DotNet'
-  inputs:
-    packageType: 'sdk'
-    version: '3.x'
-
-- task: DotNetCoreCLI@2
-  displayName: 'DotNet Build'
-  inputs:
-    command: 'build'
-    projects: $(projects)
-    arguments: '--output $(Build.ArtifactStagingDirectory) $(styleCopEnabled)'
-
-- task: DotNetCoreCLI@2
-  displayName: 'DotNet Tests'
-  inputs:
-    command: test
-    projects:  $(testsProjects)
-
-- task: PublishBuildArtifacts@1
-  displayName: 'DotNet Drop'
-  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'Drop'
-    publishLocation: 'Container'
-  
\ No newline at end of file
+variables:
+  projects: '**/*.sln'
+  azureSubscription: 'devlevel3-dev (a797242d-ce60-41aa-98ca-2e22d5221cfe)'
+  functionAppName: 'fa-tuapi-resumes-dev-bra'
+  vmImageName: 'ubuntu-latest'
+
+stages:
+- stage: Build
+  displayName: Build & Test
+
+  jobs:
+  - job: Build
+    displayName: BuildJob
+
+    pool:
+      vmImage: $(vmImageName)
+
+    steps:
+    - task: UseDotNet@2
+      displayName: 'Install DotNet'
+      inputs:
+        packageType: 'sdk'
+        version: '3.x'
+
+    - task: DotNetCoreCLI@2
+      displayName: 'DotNet Build'
+      inputs:
+        command: 'build'
+        projects: $(projects)
+        arguments: '--output $(Build.ArtifactStagingDirectory) $(styleCopEnabled)'
+
+    - task: DotNetCoreCLI@2
+      displayName: 'DotNet Tests'
+      inputs:
+        command: test
+        projects:  $(testsProjects)
+    - task: ArchiveFiles@2
+      inputs:
+        rootFolderOrFile: '$(Build.ArtifactStagingDirectory)'
+        includeRootFolder: false
+        archiveType: zip
+        archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+        replaceExistingArchive: true
+      
+    - task: PublishBuildArtifacts@1
+      displayName: 'Dotnet Drop Artifacts'
+      # condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip'
+        ArtifactName: 'drop'
+        publishLocation: 'Container'
+
+- stage: Deploy
+  displayName: Deploy stage
+  dependsOn: Build
+  condition: succeeded()
```

Comment:
make it work **only for** develop and master merges, we don't want deploy everytime

Reply:
It's done